proc/threads.c

```diff
@@ -841,6 +876,97 @@ int proc_threadCreate(process_t *process, void (*start)(void *), unsigned int *i
 }
 
 
+static unsigned int _proc_lockGetPriority(lock_t *lock);
+
+
+static unsigned int _proc_threadGetPriority(thread_t *thread)
+{
+	unsigned int ret, priority = thread->priorityBase;
+	lock_t *lock;
+	spinlock_ctx_t sc;
+
+	hal_spinlockSet(&thread->spinlock, &sc);
+
+	lock = thread->locks;
+	if (lock != NULL) {
+		do {
+			ret = _proc_lockGetPriority(lock);
+			if (ret < priority) {
+				priority = ret;
+			}
+			lock = lock->next;
+		} while (lock != thread->locks);
+	}
+
+	hal_spinlockClear(&thread->spinlock, &sc);
+
+	return priority;
+}
+
+
+static void _proc_threadSetPriority(thread_t *thread, unsigned int priority)
+{
+	unsigned int i;
+
+	for (i = 0; i < hal_cpuGetCount(); i++) {
+		if (thread == threads_common.current[i]) {
+			break;
+		}
+	}
+
+	if ((i == hal_cpuGetCount()) && (thread->state == READY)) {
+		LIST_REMOVE(&threads_common.ready[thread->priority], thread);
+		LIST_ADD(&threads_common.ready[priority], thread);
+	}
```

Comment:
Do this whole block under if (thread->state == READY) - minor performance improvement

Reply:
Done.